Core/src/IfcGeometryConverter/CurveConverter.h

```diff
@@ -678,8 +678,7 @@ namespace OpenInfraPlatform {
 					double xDeltaA = arcMid2D.x - arcStart2D.x;
 					double yDeltaB = arcEnd2D.y - arcMid2D.y;
 					double xDeltaB = arcEnd2D.x - arcMid2D.x;
-
-					if (xDeltaA != 0. && xDeltaB != 0.) {
+					if (!GeomSettings()->areEqual(xDeltaA, 0.) && !GeomSettings()->areEqual(xDeltaB, 0.)){

```

Comment:
Here it is unnecessary to check if the values are "close to" zero - they really just must not be zero. So while such change is welcomed, here it is out of order. Please return to the original.

Reply:
Resolved: 6e54eae